src/nodedef.cpp

```diff
@@ -163,6 +179,14 @@ void NodeBox::deSerialize(std::istream &is)
 		READBOXES(connect_left);
 		READBOXES(connect_back);
 		READBOXES(connect_right);
+		READBOXES(disconnected_top);
```

Comment:
Either needs a NodeBox serialization version increase or a check whether the stream hasn't reached the end yet. Just reading outside the original size is dangerous.